### PR TITLE
BUG: misc: Fix the Pillow work-around for 1-bit images.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Pillow has a bug that can cause a seg. fault when a 1-bit
image is read and converted to a numpy array.  Previously,
code was added to convert an Image instance to 8-bit before
converting it to a numpy array, but this change didn't correctly
convert the image to 8-bit (the convert method does not operate
in-place).

<strike>This change also ensures that when this work-around is used
the array returned has dtype np.uint8 with values 0 and 1.</strike>
To be consistent with how 2 bit and 4 bit files are read, the values in the array returned from a 1 bit image are 0 and 255.

Closes gh-2138.

### CODE REVIEW

**Comment by matthew-brett** [comment]
Warren - does python-Pillow have an issue to remind them about this?

**Comment by WarrenWeckesser** [comment]
@matthew-brett: Here's the pillow github issue: https://github.com/python-pillow/Pillow/issues/350

**Comment by matthew-brett** [comment]
Great - thanks - sorry to be lazy in not looking that up.

**Comment by WarrenWeckesser** [comment]
Update: returning values 0 and 1 for an image with bit depth 1 is not consistent with, for example, the behavior with bit depth 2.  When the bit depth is 2, the values are also scaled up from {0, 1, 2, 3} to {0, 85, 170, 255}.

I'll add WIP to the PR title.  This needs more investigation.

**Comment by WarrenWeckesser** [comment]
In the updated PR, I removed that code that attempted to "fix" the values returned by the PIL/Pillow code for a 1 bit image.   The values in the array returned by `imread()` when it reads a 1 bit image are 0 and 255.  This behavior is inherited from PIL/Pillow, and is consistent with how 2 bit and 4 bit images are treated (i.e. the values are scaled up to the range 0-255).  There are new tests for 1 bit, 2 bit and 4 bit grayscale images.

**Comment by rgommers** [comment]
Thanks Warren, looks good to me. That's a pretty long history for four measly missing characters (``im =``).

**Comment by rgommers** [comment]
Returning 0 and 255 makes sense to me.

**Comment by rgommers** [comment]
We've got to fix the mess with ``imread``s floating around soon-ish. We've got one in ``misc``, one in ``ndimage`` and one in ``skimage``....

**Comment by WarrenWeckesser** [comment]
*Grrr*, I forgot about `ndimage.imread`.

I'd like to deprecate `ndimage.imread`.  We could add a `mode` argument to `misc.imread`, so the implementation of `ndimage.imread` can be replaced with that of `misc.imread` during the deprecation cycle.

There are bigger API questions (e.g. Completely defer to `skimage`? What to do with `misc.im*`?), but I think they should be discussed on the mailing list.

**Comment by rgommers** [comment]
> I'd like to deprecate ``ndimage.imread``.

I'd like to deprecate all of ``misc``. And move ``pilutil`` over to ``ndimage``.

For this PR, I think it's OK to either add the same fix to ``ndimage`` (least amount of work) or to merge the two implementations, put them in a common place and import that in both ``misc`` and ``ndimage``.

**Comment by WarrenWeckesser** [comment]
> I'd like to deprecate all of `misc`.

+1

> And move pilutil over to `ndimage`.

Something like that might work.  On the other hand, it would make sense to have `imread` and `imsave` in `io`. 

>  merge the two implementations, put them in a common place and import that in both `misc` and `ndimage`.

I'll work on this today.

**Comment by rgommers** [comment]
Moving it to ``io`` does make sense. 

The other functions in ``pilutil`` aren't very interesting - they could also be deprecated instead of moved.

**Comment by WarrenWeckesser** [comment]
I haven't forgotten about this--I'll get back to it as soon as I can.  At first glance, the failure with python 2.6 is a bit mysterious, but I haven't looked too deeply.  The changes worked fine in my local installation of python 2.6.

File: scipy/ndimage/io.py [hunk 1]
```diff
@@ -1,49 +1,31 @@
...
-    return result
+    if _have_pil:
+        return _imread(fname, flatten, mode)
+    raise ImportError("Could not import the Python Imaging Library (PIL)"
+                      " required to load image files.  Please refer to"
+                      " http://pypi.python.org/pypi/PIL/ for installation"
+                      " instructions.")
+
+if _have_pil:
+    imread.__doc__ = _imread.__doc__.replace('name : str', 'fname : str')
```
Comment (rgommers) [inline scipy/ndimage/io.py]: This should be behind ``if imread.__doc__ is not None:``. That'll fix the TravisCI failure on the Python 2.6 build. Reason: that uses ``python -OO`` which strips docstrings.

**Comment by WarrenWeckesser** [comment]
@rgommers Thanks for pointing out the issue with `-OO` in the Python 2.6 build.

**Comment by rgommers** [comment]
LGTM now. Thanks Warren!

**Pull request merged**